### PR TITLE
Prevent using the Minio Extension on non-Linux OS

### PR DESCRIPTION
**Pull request by snazy**
Background is that S3 URLs encode the bucket name in front of the S3 target's FQDN, which works fine for `bucket-name.localhost` on Linux, but not on other OS'es. S3 requests encode the bucket name in the DNS FQDN - e.g. `my-bucket.my-minio.com`, which translates to `my-bucket.localhost` for local testing. Linux supports these wildcards out of the box, but macOS + Windows do not.

### CODE REVIEW

File: testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java [hunk 1]
```diff
@@ -35,10 +40,20 @@
...
   private static final ExtensionContext.Namespace NAMESPACE =
       ExtensionContext.Namespace.create(MinioExtension.class);
 
+  @Override
+  public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext context) {
+    if (OS.current() == OS.LINUX) {
+      return enabled("Running on Linux");
+    }
+    return disabled(
+        "Not running on Linux, because neither macOS nor Windows support wildcard localhost FQDNs.");
```
Comment (dimas-b) [inline testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: Hmmm... I'm not sure I follow the meaning of this message :thinking:

Comment (snazy) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: `foobar.localhost` cannot be resolved

Comment (dimas-b) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: If we disable Minio tests on Mac+Win, how could we make sure our S3-related code still works there?

Comment (dimas-b) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: (to clarify: I'm fine with this change as long as we fix the message, but I wonder if we actually need some other test env. for S3 on Mac/Win)

Comment (snazy) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: They already do not work on those platforms, this change only let's CI pass on those platforms (see #5705).
The _mocked_ S3 impl is still exercised - Minio is not.

Comment (dimas-b) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: Could we add a local host wildcards in a CI action?

Comment (snazy) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: Too complicated and doesn't work - already tried.

Comment (dimas-b) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: ok

Comment (dimas-b) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: Could you just fix the "disabled" message? It still looks awkward :)

Comment (snazy) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: It's a wildcard fully-qualified-domain-name?

Comment (dimas-b) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: No, why `not running on Linux` when the OS is _not_ Linux?

Comment (dimas-b) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: I'd say `Disabled on a non-Lixus OS, because....`

Comment (dimas-b) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: I tend to interpret the disabled message as belonging to the test :) so in my mind  `the test is` `not running on Linux, becase...` looks odd :)

Comment (snazy) [reply testing/s3minio/src/main/java/org/projectnessie/minio/MinioExtension.java:54]: Proposal?

**Review by dimas-b — APPROVED** [review]

**Pull request merged**